tests/testthat/test-slice-chop.R

```diff
@@ -307,6 +307,22 @@ test_that("NULLs are ignored when unchopped with other vectors", {
   expect_identical(list_unchop(list("a", NULL, "b"), list(2, integer(), 1)), c("b", "a"))
 })
 
+test_that("can use a `NULL` element with a corresponding index", {
+  # FIXME: Probably not quite right, but not entirely clear what it should be:
+  # - Maybe `unspecified(2)`?
+  # - Or should `NULL`s even be allowed in `list_unchop()`?
+  expect_null(list_unchop(list(NULL), indices = list(1:2)))
```

Comment:
Not entirely sure about this result, but it seems rare enough to not care too much right now

Reply:
hmm I tend to think we should respect the invariant that the output size is equal to the indices size.

You mean an unclassed unspecified vector right? That seems reasonable.

Reply:
I opened https://github.com/r-lib/vctrs/issues/1702 because I think this is also tied to what we should return in the empty case, like `vec_c()` with no inputs and `list_unchop(list(), indices = list())`